weatherman.py

```diff
@@ -0,0 +1,176 @@
+import argparse
+import calendar
+import csv
+import glob
+import os
+import re
+import statistics
+from colorama import Fore, Style
+
+
+def validate_input_date(date_input):
+    pattern = r'^\d{4}/(0[1-9]|1[0-2])$'
+    if re.match(pattern, date_input):
+        return True
+    return False
+
+
+def convert_month(weather_month_input):
+    month_number = int(weather_month_input)
+    if 1 <= month_number <= 12:
+        return calendar.month_abbr[month_number]
+
+
+def receive_weather_file_names(BASE_DIRECTORY, yearly_temperatures, weather_month=None):
+    pattern = f"*{yearly_temperatures}*"
+    if weather_month:
+        pattern = f"*{yearly_temperatures}*{weather_month}*"
+
+    weatherman_file_names = glob.glob(os.path.join(BASE_DIRECTORY, pattern))
+    return weatherman_file_names
+
+
+def read_weather_files(yearly_temperatures, weather_month, BASE_DIRECTORY):
+    weather_records = []
+
+    weatherman_file_names = receive_weather_file_names(BASE_DIRECTORY, yearly_temperatures, weather_month)
```

Comment:
Please use this outside the function before its call and pass the file names in `read_weather_files` function